src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala

```diff
@@ -0,0 +1,121 @@
+package com.advancedtelematic.campaigner.db
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.stream.Materializer
+import akka.testkit.TestKit
+import cats.data.NonEmptyList
+import com.advancedtelematic.campaigner.data.DataType._
+import com.advancedtelematic.campaigner.data.Generators._
+import com.advancedtelematic.campaigner.util.CampaignerSpecUtil
+import com.advancedtelematic.campaigner.util.DatabaseUpdateSpecUtil
+import com.advancedtelematic.libats.test.DatabaseSpec
+import org.scalacheck.Gen
+import org.scalatest._
+import org.scalatest.concurrent.ScalaFutures
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import slick.jdbc.MySQLProfile.api._
+
+final class CampaignStatusRecalculateSpec
+    extends TestKit(ActorSystem("CampaignStatusRecalculateSpec"))
+    with AsyncFlatSpecLike
```

Comment:
If you really like this style, keep it, but this leads to rather weird testing, for comprehensions everywhere, beating non future values into futures, and always having to end a future with an assertion...

Reply:
TBH I just spotted `AsyncFlatSpec` usage in one of the tests and just followed it without much research. Later I've discovered `ScalaFutures` mixin which looks much better IMO. I'll refactor this test.